syft/frameworks/torch/tensors/native.py

```diff
@@ -93,6 +94,8 @@ def create_pointer(
             ptr_id: A string or integer parameter to specify the id of the pointer
                 in case you wish to set it manually for any special reason.
                 Otherwise, it will be set randomly.
+            garbage_location: If true (default), delete the remote tensor when the
```

Comment:
What situations would we not want to delete the tensor when the pointer is deleted?
I can definitely see the case where there are multiple clients that want to access the same data on a worker but would that data be sent from a client or simply added on worker startup? I also think this parameter should be renamed to something like garbage_collect_data

Reply:
This is because of the way we send data.
In the previous version of pysyft we had a `private` attribute in the serialisation process to desambiguate two scenarios:
- *local* send a tensor to *bob* which must receive the tensor and not a pointer to it (send with `private = false`)
- *local* performs an op on a pointer at *bob*, and must receive a pointer, so bob send back the result with `private = true` (and in practice then the tensor is serialized without its data)

What we do instead is now for the case non-private to just pickle/ser the tensor and send it, and for the private case to create locally the pointer and send it (see base.py execute_command), partly because the built-in method for ser torch tensor keep all the data.

But of course if *bob* creates a pointer to his own result, and sends it back to *local*, then this pointer gets garbage collected and the result is deleted, which we want to avoid. That's why we add a property `dont_garbage_collect_target` (which is not serialized and sent to local)

Reply:
Agreed on the `garbage_collect_data` name